pkg/controller/migmigration/stage.go

```diff
@@ -0,0 +1,415 @@
+package migmigration
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	migref "github.com/konveyor/mig-controller/pkg/reference"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StagePod - wrapper for stage pod, allowing to compare  two stage pods for equality
+type StagePod struct {
+	corev1.Pod
+}
+
+// StagePodList - a list of stage pods, with built-in stage pod deduplication
+type StagePodList []StagePod
+
+// BuildStagePods - creates a list of stage pods from a list of pods
+func BuildStagePods(list *corev1.PodList) StagePodList {
+	stagePods := StagePodList{}
+	for _, pod := range list.Items {
+		stagePods.merge(buildStagePodFromPod(migref.ObjectKey(&pod), pod.Labels, &pod))
+	}
+	return stagePods
+}
+
+func (p StagePod) equals(pod StagePod) bool {
```

Comment:
This equals method only confirms that all of the volumes from this pod's volumes are present in the passed-in volume. If p1 includes vol1 and vol2, and p2 includes vol1, vol2, and vol3, then p1.equals(p2) is true, while p2.equals(p1) is false.

It looks like this doesn't break anything, though, based on the order things are being called.